location-service/src/main/java/uk/gov/cshr/locationservice/RegionLookup.java

```diff
@@ -0,0 +1,145 @@
+
+                    Path2D.Double polygon = new Path2D.Double();
+
+                    boolean moved = false;
+
+                    Iterator<JsonNode> valuesIterator = coordinatesIterator.next().elements();
+                    while (valuesIterator.hasNext()) {
+
+                        JsonNode coordinates = valuesIterator.next();
+                        Double longitude = coordinates.get(0).asDouble();
+                        Double latitude = coordinates.get(1).asDouble();
+
+                        if (!moved) {
+                            polygon.moveTo(latitude, longitude);
+                            moved = true;
+                        }
+                        else {
+                            polygon.lineTo(latitude, longitude);
+                        }
+
+                    }
+
+                    pathsList.add(polygon);
+                }
+                else if (featureNode.get("geometry").get("type").asText().equals("MultiPolygon")) {
+
+                    Iterator<JsonNode> valuesIterator = coordinatesIterator.next().elements();
+                    while (valuesIterator.hasNext()) {
+
+                        Iterator<JsonNode> polygonIterator = valuesIterator.next().elements();
+
+                        Path2D.Double polygon = new Path2D.Double();
+
+                        boolean moved = false;
+
+                        while (polygonIterator.hasNext()) {
+
+                            JsonNode polyNode = polygonIterator.next();
+                            Double longitude = polyNode.get(0).asDouble();
+                            Double latitude = polyNode.get(1).asDouble();
+
+                            if (!moved) {
+                                polygon.moveTo(latitude, longitude);
+                                moved = true;
+                            }
+                            else {
+                                polygon.lineTo(latitude, longitude);
+                            }
+                        }
+
+                        pathsList.add(polygon);
+                    }
+                }
+            }
+        }
+
+        return ukNutsMap;
+    }
+
+    public static UK_NUTS findRegion(Double latitude, Double longitude) throws RuntimeException {
+
+        try {
+
+            if (nutsMap == null) {
+                nutsMap = createNutsMap();
+            }
+
+            for (Map.Entry<UK_NUTS, List<Path2D.Double>> entry : nutsMap.entrySet()) {
+
+                UK_NUTS ukNuts = entry.getKey();
+                List<Path2D.Double> polygons = entry.getValue();
+
+                for (Path2D.Double polygon : polygons) {
+
+                    if (polygon.contains(latitude, longitude)) {
+                        return ukNuts;
+                    }
+                }
+            }
+
+            return null;
+        }
+        catch (IOException ioe) {
+            throw new RuntimeException("Exception initialising the NUTS map", ioe);
```

Comment:
I disagree.  The exception has been converted from a checked (IOException) to a runtime because it indicates something has gone wrong which is out of control of the program namely that the data file could not be read.

This is a setup issue, is non recoverable, and should be thrown up the stack.

Checked exceptions should be recoverable while runtime exceptions indicate the application is effectively broken.

"Here's the bottom line guideline: If a client can reasonably be expected to recover from an exception, make it a checked exception. If a client cannot do anything to recover from the exception, make it an unchecked exception."
https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html